error in resolve conflict

**Hi 
my ide make exception when my version and server version is in conflict

please help me**




```

java.lang.NoClassDefFoundError: Could not initialize class com.microsoft.alm.plugin.idea.tfvc.core.tfs.conflicts.ConflictsEnvironment
at com.microsoft.alm.plugin.idea.tfvc.core.TFSUpdateEnvironment.updateDirectories(TFSUpdateEnvironment.java:107)
at com.intellij.openapi.vcs.update.AbstractCommonUpdateAction$Updater.performUpdate(AbstractCommonUpdateAction.java:389)
at com.intellij.openapi.vcs.update.AbstractCommonUpdateAction$Updater.runImpl(AbstractCommonUpdateAction.java:358)
at com.intellij.openapi.vcs.update.AbstractCommonUpdateAction$Updater.run(AbstractCommonUpdateAction.java:332)
at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:436)
at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:120)
at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsync$5(CoreProgressManager.java:496)
at com.intellij.codeWithMe.ClientId$Companion$decorateFunction$2.apply(ClientId.kt:238)
at com.intellij.codeWithMe.ClientId$Companion$decorateFunction$2.apply(ClientId.kt:238)
at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:244)
at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:188)
at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:624)
at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:698)
at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:646)
at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:623)
at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:175)
at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:244)
at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
at java.base/java.security.AccessController.doPrivileged(Native Method)
at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
at java.base/java.lang.Thread.run(Thread.java:829)



java.lang.NoClassDefFoundError: com/intellij/openapi/diff/DiffRequest
	at com.microsoft.alm.plugin.idea.tfvc.core.tfs.conflicts.ConflictsEnvironment.<clinit>(ConflictsEnvironment.java:27)
	at com.microsoft.alm.plugin.idea.tfvc.core.TFSUpdateEnvironment.updateDirectories(TFSUpdateEnvironment.java:107)
	at com.intellij.openapi.vcs.update.AbstractCommonUpdateAction$Updater.performUpdate(AbstractCommonUpdateAction.java:389)
	at com.intellij.openapi.vcs.update.AbstractCommonUpdateAction$Updater.runImpl(AbstractCommonUpdateAction.java:358)
	at com.intellij.openapi.vcs.update.AbstractCommonUpdateAction$Updater.run(AbstractCommonUpdateAction.java:332)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:436)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:120)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsync$5(CoreProgressManager.java:496)
	at com.intellij.codeWithMe.ClientId$Companion$decorateFunction$2.apply(ClientId.kt:238)
	at com.intellij.codeWithMe.ClientId$Companion$decorateFunction$2.apply(ClientId.kt:238)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:244)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:188)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:624)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:698)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:646)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:623)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:175)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:244)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ClassNotFoundException: com.intellij.openapi.diff.DiffRequest PluginClassLoader(plugin=PluginDescriptor(name=Azure DevOps, id=com.microsoft.vso.idea, descriptorPath=plugin.xml, path=~\AppData\Roaming\JetBrains\Rider2021.3\plugins\com.microsoft.vso.idea, version=1.162.0, package=null, isBundled=false), packagePrefix=null, instanceId=90, state=active)
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:235)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 27 more
```

--- a/plugin/src/com/microsoft/alm/plugin/idea/tfvc/core/tfs/conflicts/DialogContentMerger.java
+++ b/plugin/src/com/microsoft/alm/plugin/idea/tfvc/core/tfs/conflicts/DialogContentMerger.java
@@ -19,15 +19,14 @@
 
 package com.microsoft.alm.plugin.idea.tfvc.core.tfs.conflicts;
 
-import com.intellij.openapi.application.ApplicationManager;
-import com.intellij.openapi.diff.ActionButtonPresentation;
-import com.intellij.openapi.diff.DiffManager;
-import com.intellij.openapi.diff.DiffRequestFactory;
-import com.intellij.openapi.diff.MergeRequest;
+import com.intellij.diff.DiffManager;
+import com.intellij.diff.DiffRequestFactory;
+import com.intellij.diff.InvalidDiffRequestException;
+import com.intellij.diff.merge.MergeRequest;
+import com.intellij.diff.merge.MergeResult;
 import com.intellij.openapi.editor.Document;
 import com.intellij.openapi.fileEditor.FileDocumentManager;
 import com.intellij.openapi.project.Project;
-import com.intellij.openapi.ui.DialogWrapper;
 import com.intellij.openapi.util.io.StreamUtil;
 import com.intellij.openapi.vcs.history.VcsRevisionNumber;
 import com.intellij.openapi.vcs.merge.MergeDialogCustomizer;
@@ -36,6 +35,9 @@ import com.microsoft.alm.common.utils.ArgumentHelper;
 import com.microsoft.alm.plugin.idea.tfvc.ui.resolve.ContentTriplet;
 
 import java.io.File;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
 
 public class DialogContentMerger implements ContentMerger {
 
@@ -43,37 +45,32 @@ public class DialogContentMerger implements ContentMerger {
                                 final VcsRevisionNumber serverVersion) {
         ArgumentHelper.checkIfFileWriteable(new File(localFile.getPath()));
 
+        Document document = Objects.requireNonNull(FileDocumentManager.getInstance().getDocument(localFile));
         final MergeDialogCustomizer c = new MergeDialogCustomizer();
-        final MergeRequest request = DiffRequestFactory.getInstance().createMergeRequest(StreamUtil.convertSeparators(contentTriplet.localContent),
-                StreamUtil.convertSeparators(contentTriplet.serverContent),
-                StreamUtil.convertSeparators(contentTriplet.baseContent),
-                localFile, project,
-                ActionButtonPresentation.APPLY,
-                ActionButtonPresentation.CANCEL_WITH_PROMPT);
-
-        request.setWindowTitle(c.getMergeWindowTitle(localFile));
-        request.setVersionTitles(new String[]{
-                c.getLeftPanelTitle(localFile),
-                c.getCenterPanelTitle(localFile),
-                c.getRightPanelTitle(localFile, serverVersion)
-        });
-
-        // TODO (JetBrains) call canShow() first
-        DiffManager.getInstance().getDiffTool().show(request);
-        if (request.getResult() == DialogWrapper.OK_EXIT_CODE) {
-            return true;
-        } else {
-            request.restoreOriginalContent();
-            // TODO (JetBrains) maybe MergeVersion.MergeDocumentVersion.restoreOriginalContent() should save document itself?
-            ApplicationManager.getApplication().runWriteAction(new Runnable() {
-                public void run() {
-                    final Document document = FileDocumentManager.getInstance().getDocument(localFile);
-                    if (document != null) {
-                        FileDocumentManager.getInstance().saveDocument(document);
-                    }
-                }
-            });
-            return false;
+        final MergeRequest request;
+        AtomicReference<MergeResult> result = new AtomicReference<>();
+        try {
+            request = DiffRequestFactory.getInstance().createMergeRequest(
+                    project,
+                    localFile.getFileType(),
+                    document,
+                    Arrays.asList(
+                        StreamUtil.convertSeparators(contentTriplet.localContent),
+                        StreamUtil.convertSeparators(contentTriplet.baseContent),
+                        StreamUtil.convertSeparators(contentTriplet.serverContent)
+                    ),
+                    c.getMergeWindowTitle(localFile),
+                    Arrays.asList(
+                            c.getLeftPanelTitle(localFile),
+                            c.getCenterPanelTitle(localFile),
+                            c.getRightPanelTitle(localFile, serverVersion)
+                    ), result::set);
+        } catch (InvalidDiffRequestException e) {
+            throw new RuntimeException(e);
         }
+
+        DiffManager.getInstance().showMerge(project, request);
+
+        return result.get() != MergeResult.CANCEL;
     }
 }

